packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx

```diff
@@ -124,12 +124,13 @@ const ActiveVisitsTable = (props) => {
                 </TableRow>
               </TableHead>
               <TableBody>
-                {rows.slice(lowerBound, upperBound).map((row) => (
+                {rows.slice(lowerBound, upperBound).map((row, ind) => (
```

Comment:
Why anyway this whole thing with `slice`? You have `usePagination`. It gives you already the subset. Use this.

Reply:
I need all of the results when I will be applying searching. That's why I pass all the activeVisits in the line 106 to the datatable and whenever search happens, the corresponding rows are output which are then sliced.

If I use `usePagination` I get the sliced data which if passed in the datatable will implement the search through that specific sliced rows only.

Reply:
No. You should have the search to look up in the activeVisits data set and supply the filtered data set to the `usePagination` hook.

Reply:
But search is handled by the dataTable. If I give it a try by myself, I'll have to iterate through all the column to find the results.

Reply:
If you want by that way, I'll implement it too. But my recommendation would be to go through the pre-built search by dataTable. Slicing by usePagination does the same thing it is implemented right now.

Reply:
This is unnecessary and (as we can see) a source of bugs. Just implement it the right away.

Reply:
It wan't a bug, it was a mistake

Reply:
Isn't almost every bug a mistake? But in any case - that's exactly my point.